Fixed an issue related to invalid characters object and material naming

Micheus opened the pull request:
NOTE: When objects and materials was including invalid charactes the .dae file couldn't be imported by other applications. Thanks to Wҽɳԃιҽ Bʅαƈƙƚԋσɾɳ

dgud commented:
I don't understand this, what characters couldn't be handled? 
And I can't use Swedish characters in my objects?  
Doesn't collada handle unicode characters?

Micheus commented:
> I don't understand this, what characters couldn't be handled? And I can't use Swedish characters in my objects? Doesn't collada handle unicode characters?

It's not this kind of invalide character, may it can be better explained as bad identifier naming. (see https://github.com/dgud/wings/pull/555#issuecomment-1726627964).

dgud commented:
But your patch removes any character that is not A-Z or Integers?

dgud commented:
From what I can see, I believe the exported xml looks good as it is, even using </> and :  characters.  And we don't have problem importing and we use the xml standard to read it,
which would complain if the xml was horrible bad.

What application was that couldn't import the collada file wings exported.

dgud commented:
You are correct colon is not allowed (and many other characters) and may not start with a number.


https://stackoverflow.com/questions/1631396/what-is-an-xsncname-type-and-when-should-it-be-used

Micheus commented:
> You are correct colon is not allowed (and many other characters) and may not start with a number.
> 
> https://stackoverflow.com/questions/1631396/what-is-an-xsncname-type-and-when-should-it-be-used

So, I should improve the validation to check all the rules for XML files or we can just let this change on the collada exporter that is using it?

This issue was reported by an user at Discord and I checked and tested it on Blender to see that was really a problem. 
It started when you modified the Text primitive and included the prefix "Text : " to the primitive name.
Then, as it seemed a minor issue I just removed that character you added and changed the exporter to prevent the user to add any strange character to the object/material name.

dgud commented:
Something like this?  So if someone using only unicode chars still get
an export that hopefully works?

```
validate_name(Name)->
    case re:replace(Name, "[^A-Z^a-z^0-9^-]+", "_", [global,{return,list}]) of
        %% Names must be uniq, check that they are not too short after previous
        %% mangling and if they are NOT create a (ugly) uniq valid name
        "" ->
            base64:encode_to_string(unicode:characters_to_binary(Name), #{mode=>urlsafe});
        "_" ->
            base64:encode_to_string(unicode:characters_to_binary(Name), #{mode=>urlsafe});
        [Int|_] when $0 =< Int, Int =< $9 ->
            base64:encode_to_string(unicode:characters_to_binary(Name), #{mode=>urlsafe});
        HopefullyUniqueString ->
            HopefullyUniqueString
    end.
```

dgud commented:
I do think it is only collada that have a problem hopefully.

Micheus commented:
> I do think it is only collada that have a problem hopefully.

I think that too.
I'm going to use the change you suggested.
